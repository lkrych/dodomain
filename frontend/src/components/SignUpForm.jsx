import React, {Component} from 'react';

class LoginForm extends Component {
  constructor(props) {
    super(props);
    this.state =  {
      errors: '',
      email: '',
      password: '',
      password_confirmation: '',
    };
    this.onInput = this.onInput.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }
  //update state fields
  onInput (property) {
    return (e) => {
      this.setState({ [property]: e.target.value });
    };
  }

  onSubmit(e) {
    e.preventDefault();
    if (!this.state.email) {
      return this.setState({
        errors: 'Email is required.',
        message: undefined,
        disableSubmit: false
      });
    }
    if (this.state.password !== this.state.password_confirmation) {
      return this.setState({
        errors: 'Passwords must match.',
        message: undefined,
        disableSubmit: false
      });
    }
    if (this.state.password.length < 6 || this.state.password.length > 20 ) {
      return this.setState({
        errors: 'Passwords should be at least 6 characters and have fewer than 20 characters.',
        message: undefined,
        disableSubmit: false
      });
    }
    this.props.signUp({
      email: this.state.email,
      password: this.state.password,
    });
  }

  componentWillUnmount(){
    this.props.clearErrors();
  }

  render() {
    let textErrors = <div className="errors">{this.props.errors}</div>;
    return (
      <div>
        {textErrors}
        <form>
        <div className='field'>
            <label> Email </label>
            <br />
            <input autoFocus='autofocus' type='email'
              id='email' 
              onChange={this.onInput('email')}
              value={this.state.email} />
          </div>
          <div className='field' >
            <label> Password </label>
            <br />
            <input autoComplete='off' type='password'
              id='password' 
              onChange={this.onInput('password')}
              value={this.state.password}
                />
            </div>

          <div className='field' >
            <label> Password Confirmation </label>
            <br />
            <input autoComplete='off' type='password'
              id='password_confirmation' 
              onChange={this.onInput('password_confirmation')}
              value={this.state.password_confirmation}
                />
            </div>

          < input
            type="submit"
            className="btn btn-primary btn-submit"
            onClick={this.onSubmit}/>
        </form>
      </div>
    );
  }
}

export default LoginForm;
