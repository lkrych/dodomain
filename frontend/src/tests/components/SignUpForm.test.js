import React from 'react';
import { shallow } from 'enzyme';

import SignUpForm from '../../components/SignUpForm';

describe('SignUpForm', () => {
  const mockSubmitForm = jest.fn();
  const signUpForm = shallow(<SignUpForm signUp={mockSubmitForm} classes={{root:"blahStyle"}} />);
  
  it('renders correctly', () => {
    expect(signUpForm).toMatchSnapshot();
  });

  it('correctly initializes the `state` with an empty email, password, password_confirmation, and errors', () => {
    expect(signUpForm.state().email).toEqual('');
    expect(signUpForm.state().password).toEqual('');
    expect(signUpForm.state().password_confirmation).toEqual('');
    expect(signUpForm.state().errors).toEqual('');
  });

  describe('when typing into the email input', () => {
    const email = 'Noodle@gmail.com';

    beforeEach(() => {
      signUpForm.find('#email').simulate('change', { target: { value: email} });
    });

    it('updates the person in `state`', () => {
      expect(signUpForm.state().email).toEqual(email);
    });
  });

  describe('when typing into the password input', () => {
    const password = 'password';

    beforeEach(() => {
      signUpForm.find('#password').simulate('change', { target: { value: password} });
    });

    it('updates the person in `state`', () => {
      expect(signUpForm.state().password).toEqual(password);
    });
  });

  describe('when typing into the password_confirmation input', () => {
    const password_confirmation = 'password';

    beforeEach(() => {
      signUpForm.find('#password_confirmation').simulate('change', { target: { value: password_confirmation} });
    });

    it('updates the person in `state`', () => {
      expect(signUpForm.state().password_confirmation).toEqual(password_confirmation);
    });
  });

  describe('on submission of form', () => {
    beforeEach(() => {
      signUpForm.find('#email').simulate('change', { target: { value: ''} });
      signUpForm.find('#password').simulate('change', { target: { value: ''} });
      signUpForm.find('#password_confirmation').simulate('change', { target: { value: ''} });
    });

    const mockedEvent = { target: {}, preventDefault: jest.fn() };
    it('should populate the errors `state` if the user tries to submit without an email', () => {
      signUpForm.find('.btn-submit').simulate('click', mockedEvent);
      expect(signUpForm.state().errors).toEqual('Email is required.');
    });

    it('should populate the errors `state` if the passwords dont match', () => {
      signUpForm.find('#email').simulate('change', { target: { value: 'test@email.com'} });
      signUpForm.find('#password').simulate('change', { target: { value: 'password'} });
      signUpForm.find('#password_confirmation').simulate('change', { target: { value: 'passwood'} });
      signUpForm.find('.btn-submit').simulate('click', mockedEvent);

      expect(signUpForm.state().errors).toEqual('Passwords must match.');

    });

    it('should populate the errors state if the submitted password is too short', () => {
      signUpForm.find('#email').simulate('change', { target: { value: 'test@email.com'} });
      signUpForm.find('#password').simulate('change', { target: { value: 'pass'} });
      signUpForm.find('#password_confirmation').simulate('change', { target: { value: 'pass'} });
      signUpForm.find('.btn-submit').simulate('click', mockedEvent);

      expect(signUpForm.state().errors).toEqual("Passwords should be at least 6 characters and have fewer than 20 characters.");
    });

    it('should populate the errors state if the submitted password is too long', () => {
      signUpForm.find('#email').simulate('change', { target: { value: 'test@email.com'} });
      signUpForm.find('#password').simulate('change', { target: { value: 'reallyreallyreallylongpassword'} });
      signUpForm.find('#password_confirmation').simulate('change', { target: { value: 'reallyreallyreallylongpassword'} });
      signUpForm.find('.btn-submit').simulate('click', mockedEvent);

      expect(signUpForm.state().errors).toEqual("Passwords should be at least 6 characters and have fewer than 20 characters.");
    });
  });

  describe('Error Text', () => {
    beforeAll(() => { //reset error state
      signUpForm.state.errors = '';
    });

    it('should be an empty string if there are no session_errors', () => {
      expect(signUpForm.find('#errors').html()).not.toContain("there is an error");
    });
    
    let signUpFormWithErrors = shallow(<SignUpForm signUp={mockSubmitForm} errors={"there is an error"} classes={{root:"blahStyle"}}/>);
    it('should be populated if there are session_errors', () => {
      expect(signUpFormWithErrors.find('#errors').html()).toContain("there is an error");
    });
  });

});