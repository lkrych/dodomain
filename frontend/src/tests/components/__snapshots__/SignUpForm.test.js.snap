// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpForm renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignUpForm
    classes={
        Object {
            "root": "blahStyle",
          }
    }
    signUp={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <WithStyles(Typography)
          color="error"
          id="errors"
          variant="body2"
/>,
        <WithStyles(Typography)
          gutterBottom={true}
          variant="headline"
>
          Sign Up
</WithStyles(Typography)>,
        <form>
          <div
                    className="field"
          >
                    <TextField
                              error={null}
                              id="email"
                              label="email"
                              onChange={[Function]}
                              required={false}
                              select={false}
                              type="email"
                              value=""
                    />
          </div>
          <div
                    className="field"
          >
                    <TextField
                              autoComplete="off"
                              error={null}
                              id="password"
                              label="password"
                              onChange={[Function]}
                              required={false}
                              select={false}
                              type="password"
                              value=""
                    />
          </div>
          <div
                    className="field"
          >
                    <TextField
                              autoComplete="off"
                              error={null}
                              id="password_confirmation"
                              label="password confirmation"
                              onChange={[Function]}
                              required={false}
                              select={false}
                              type="password"
                              value=""
                    />
          </div>
          <br />
          <WithStyles(Button)
                    className="btn btn-primary btn-submit"
                    color="primary"
                    onClick={[Function]}
                    type="submit"
                    variant="contained"
          >
                    Sign Up
          </WithStyles(Button)>
</form>,
      ],
      "className": "blahStyle",
      "elevation": 4,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "",
          "color": "error",
          "id": "errors",
          "variant": "body2",
        },
        "ref": null,
        "rendered": "",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Sign Up",
          "gutterBottom": true,
          "variant": "headline",
        },
        "ref": null,
        "rendered": "Sign Up",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="field"
>
              <TextField
                            error={null}
                            id="email"
                            label="email"
                            onChange={[Function]}
                            required={false}
                            select={false}
                            type="email"
                            value=""
              />
</div>,
            <div
              className="field"
>
              <TextField
                            autoComplete="off"
                            error={null}
                            id="password"
                            label="password"
                            onChange={[Function]}
                            required={false}
                            select={false}
                            type="password"
                            value=""
              />
</div>,
            <div
              className="field"
>
              <TextField
                            autoComplete="off"
                            error={null}
                            id="password_confirmation"
                            label="password confirmation"
                            onChange={[Function]}
                            required={false}
                            select={false}
                            type="password"
                            value=""
              />
</div>,
            <br />,
            <WithStyles(Button)
              className="btn btn-primary btn-submit"
              color="primary"
              onClick={[Function]}
              type="submit"
              variant="contained"
>
              Sign Up
</WithStyles(Button)>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <TextField
                error={null}
                id="email"
                label="email"
                onChange={[Function]}
                required={false}
                select={false}
                type="email"
                value=""
/>,
              "className": "field",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "error": null,
                "id": "email",
                "label": "email",
                "onChange": [Function],
                "required": false,
                "select": false,
                "type": "email",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <TextField
                autoComplete="off"
                error={null}
                id="password"
                label="password"
                onChange={[Function]}
                required={false}
                select={false}
                type="password"
                value=""
/>,
              "className": "field",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "autoComplete": "off",
                "error": null,
                "id": "password",
                "label": "password",
                "onChange": [Function],
                "required": false,
                "select": false,
                "type": "password",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <TextField
                autoComplete="off"
                error={null}
                id="password_confirmation"
                label="password confirmation"
                onChange={[Function]}
                required={false}
                select={false}
                type="password"
                value=""
/>,
              "className": "field",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "autoComplete": "off",
                "error": null,
                "id": "password_confirmation",
                "label": "password confirmation",
                "onChange": [Function],
                "required": false,
                "select": false,
                "type": "password",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Sign Up",
              "className": "btn btn-primary btn-submit",
              "color": "primary",
              "onClick": [Function],
              "type": "submit",
              "variant": "contained",
            },
            "ref": null,
            "rendered": "Sign Up",
            "type": [Function],
          },
        ],
        "type": "form",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <WithStyles(Typography)
            color="error"
            id="errors"
            variant="body2"
/>,
          <WithStyles(Typography)
            gutterBottom={true}
            variant="headline"
>
            Sign Up
</WithStyles(Typography)>,
          <form>
            <div
                        className="field"
            >
                        <TextField
                                    error={null}
                                    id="email"
                                    label="email"
                                    onChange={[Function]}
                                    required={false}
                                    select={false}
                                    type="email"
                                    value=""
                        />
            </div>
            <div
                        className="field"
            >
                        <TextField
                                    autoComplete="off"
                                    error={null}
                                    id="password"
                                    label="password"
                                    onChange={[Function]}
                                    required={false}
                                    select={false}
                                    type="password"
                                    value=""
                        />
            </div>
            <div
                        className="field"
            >
                        <TextField
                                    autoComplete="off"
                                    error={null}
                                    id="password_confirmation"
                                    label="password confirmation"
                                    onChange={[Function]}
                                    required={false}
                                    select={false}
                                    type="password"
                                    value=""
                        />
            </div>
            <br />
            <WithStyles(Button)
                        className="btn btn-primary btn-submit"
                        color="primary"
                        onClick={[Function]}
                        type="submit"
                        variant="contained"
            >
                        Sign Up
            </WithStyles(Button)>
</form>,
        ],
        "className": "blahStyle",
        "elevation": 4,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "",
            "color": "error",
            "id": "errors",
            "variant": "body2",
          },
          "ref": null,
          "rendered": "",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Sign Up",
            "gutterBottom": true,
            "variant": "headline",
          },
          "ref": null,
          "rendered": "Sign Up",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="field"
>
                <TextField
                                error={null}
                                id="email"
                                label="email"
                                onChange={[Function]}
                                required={false}
                                select={false}
                                type="email"
                                value=""
                />
</div>,
              <div
                className="field"
>
                <TextField
                                autoComplete="off"
                                error={null}
                                id="password"
                                label="password"
                                onChange={[Function]}
                                required={false}
                                select={false}
                                type="password"
                                value=""
                />
</div>,
              <div
                className="field"
>
                <TextField
                                autoComplete="off"
                                error={null}
                                id="password_confirmation"
                                label="password confirmation"
                                onChange={[Function]}
                                required={false}
                                select={false}
                                type="password"
                                value=""
                />
</div>,
              <br />,
              <WithStyles(Button)
                className="btn btn-primary btn-submit"
                color="primary"
                onClick={[Function]}
                type="submit"
                variant="contained"
>
                Sign Up
</WithStyles(Button)>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <TextField
                  error={null}
                  id="email"
                  label="email"
                  onChange={[Function]}
                  required={false}
                  select={false}
                  type="email"
                  value=""
/>,
                "className": "field",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "error": null,
                  "id": "email",
                  "label": "email",
                  "onChange": [Function],
                  "required": false,
                  "select": false,
                  "type": "email",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <TextField
                  autoComplete="off"
                  error={null}
                  id="password"
                  label="password"
                  onChange={[Function]}
                  required={false}
                  select={false}
                  type="password"
                  value=""
/>,
                "className": "field",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "autoComplete": "off",
                  "error": null,
                  "id": "password",
                  "label": "password",
                  "onChange": [Function],
                  "required": false,
                  "select": false,
                  "type": "password",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <TextField
                  autoComplete="off"
                  error={null}
                  id="password_confirmation"
                  label="password confirmation"
                  onChange={[Function]}
                  required={false}
                  select={false}
                  type="password"
                  value=""
/>,
                "className": "field",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "autoComplete": "off",
                  "error": null,
                  "id": "password_confirmation",
                  "label": "password confirmation",
                  "onChange": [Function],
                  "required": false,
                  "select": false,
                  "type": "password",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Sign Up",
                "className": "btn btn-primary btn-submit",
                "color": "primary",
                "onClick": [Function],
                "type": "submit",
                "variant": "contained",
              },
              "ref": null,
              "rendered": "Sign Up",
              "type": [Function],
            },
          ],
          "type": "form",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
